### PR TITLE
Move StringManager to its own component

### PR DESCRIPTION
**Pull request by dimitry-**
Test: build, run StringManagerTests

### CODE REVIEW

**Comment by dpallotti** [comment]
To get some more context:
I assume this change is because the class will be used by multiple modules?
And do we not want this class in `OrbitBase`? What would be the downsides of that?

**Comment by dimitry-** [comment]
> To get some more context:
> I assume this change is because the class will be used by multiple modules?
Yes, it is going to be use in CaptureDataProvider

> And do we not want this class in `OrbitBase`? What would be the downsides of that?
We will probably end up removing it at the end of the 2nd stage. This is why I did not want to move it to OrbitBase

**Comment by dpallotti** [comment]
> We will probably end up removing it at the end of the 2nd stage. This is why I did not want to move it to OrbitBase

Sounds good, but then I would suggest a TODO and a bug explaining when the class should be removed (or moved to `OrbitBase` if it can't be removed at the time you assume it would be removed).

**Review by dpallotti — APPROVED** [review]

**Comment by dimitry-** [comment]
> > We will probably end up removing it at the end of the 2nd stage. This is why I did not want to move it to OrbitBase
> 
> Sounds good, but then I would suggest a TODO and a bug explaining when the class should be removed (or moved to `OrbitBase` if it can't be removed at the time you assume it would be removed).

It seems to be a bit early for a TODO... todo is something that could probably be addressed independently. This is part of long capture project and cannot be done before we finish with 2nd stage.

**Comment by dpallotti** [comment]
> It seems to be a bit early for a TODO

Why do you think it's early? It would explain what you just explained to me, but we would have a reference with a bug and in the code instead of just in a code review.

**Comment by dimitry-** [comment]
we are going to end up doing this for CaptureData, DataManager and some other number of classes. I think this is not a todo for the current state of the project. If we end up not removing them but refactoring them this todo will prove itself invalid. too early to tell.

"TODO: This will probably be removed at some point in the future" applies to almost everything :)

File: src/OrbitGl/ManualInstrumentationManager.h [hunk 1]
```diff
@@ -37,8 +37,7 @@ class ManualInstrumentationManager {
 
  private:
   absl::flat_hash_set<AsyncTimerInfoListener*> async_timer_info_listeners_;
-  absl::flat_hash_map<uint32_t, orbit_client_protos::TimerInfo> async_timer_info_start_by_id_;
```
Comment (florian-kuebler) [inline src/OrbitGl/ManualInstrumentationManager.h:40]: Why are you removing this line?

**Comment by dpallotti** [comment]
> we are going to end up doing this for CaptureData, DataManager and some other number of classes. I think this is not a todo for the current state of the project. If we end up not removing them but refactoring them this todo will prove itself invalid. too early to tell.
> 
> "TODO: This will probably be removed at some point in the future" applies to almost everything :)

Ok, from what you had said I had understood that the plan on when to remove this was clearer than apparently is.

File: src/OrbitGl/ManualInstrumentationManager.h (hunk 1 above)
Comment (dpallotti) [reply src/OrbitGl/ManualInstrumentationManager.h:40]: It's unused.

Comment (florian-kuebler) [reply src/OrbitGl/ManualInstrumentationManager.h:40]: Right, sorry I confused that field with the similar one in ApiEventProcessor.

Comment (dimitry-) [reply src/OrbitGl/ManualInstrumentationManager.h:40]: nobody uses this field

**Review by florian-kuebler — APPROVED** [review]

**Pull request merged**